units.cpp

```diff
@@ -1,10 +1,21 @@
 #include "units.h"
 #include <fstream>
+#include <cmath>
 
-Unit::Unit(std::string name, int hp, int damage) :m_name(name), m_hp(hp), m_damage(damage) {}
+Unit::Unit(const std::string& name, int hp, int damage) :m_name(name), m_hp(hp), m_damage(damage), m_xp(0), m_level(1), m_maxHP(m_hp) {}
 
 void Unit::dealDamageTo(Unit& damagedUnit){
-	damagedUnit.m_hp = damagedUnit.getHp() - this->getDamage() < 0 ? 0 : damagedUnit.getHp() - this->getDamage();
+	int damageDone = getDamage() > damagedUnit.getHp() ? damagedUnit.getHp(): getDamage();

```

Comment:
ez nem tűnik jónak

Reply:
Itt azért írtam így, hogyha nagyobb a damage mint a hp, akkor nem sebez többet csak amennyi a hp-ja az enemynek, egyébként meg ha kevesebb marad a sima dmg

Reply:
lehet nem vettem észre, hogy most már nem a hp-nek lesz beállítva a kifejezés eredménye, hanem abból lesz levonva, de jó így